### PR TITLE
nRF52: add support for RTC event handling

### PR DESCRIPTION
**Pull request by v01d**
## Summary

Adds support for setting up events on RTC peripheral

## Impact

Add support for new feature.

## Testing

Used events and work correctly.

### CODE REVIEW

File: arch/arm/src/nrf52/nrf52_rtc.h [hunk 1]
```diff
@@ -74,6 +76,18 @@ enum nrf52_rtc_irq_e
   NRF52_RTC_INT_COMPARE3 = 5,
 };
 
+/* RTC Events */
```
Comment (raiden00pl) [inline arch/arm/src/nrf52/nrf52_rtc.h:79]: Don't you think we should combine this with 'enum nrf52_rtc_irq_e' defined above ?
They carry the same information and there is no point in duplicating them.

Comment (v01d) [reply arch/arm/src/nrf52/nrf52_rtc.h:79]: If I do that I risk breaking code which used those definitions. Not sure what is preferable.

Comment (raiden00pl) [reply arch/arm/src/nrf52/nrf52_rtc.h:79]: If that breaks someone's compilation, fixing it is trivial.
I think the name 'nrf52_rtc_evt_e' seems more appropriate.

Comment (v01d) [reply arch/arm/src/nrf52/nrf52_rtc.h:79]: So you suggest removing the irq enum, leave the evt one and use that, right?

Comment (raiden00pl) [reply arch/arm/src/nrf52/nrf52_rtc.h:79]: > So you suggest removing the irq enum, leave the evt one and use that, right?

Exactly.

Comment (v01d) [reply arch/arm/src/nrf52/nrf52_rtc.h]: done

**Review by raiden00pl — APPROVED** [review]
LGTM

**Pull request merged**